nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -159,11 +161,42 @@ public abstract class AbstractMongoProcessor extends AbstractProcessor {
     static final List<PropertyDescriptor> descriptors;
 
     static {
-        List<PropertyDescriptor> _temp = new ArrayList<>();
-        _temp.add(CLIENT_SERVICE);
-        _temp.add(DATABASE_NAME);
-        _temp.add(COLLECTION_NAME);
-        descriptors = Collections.unmodifiableList(_temp);
+      descriptors = List.of(CLIENT_SERVICE, DATABASE_NAME, COLLECTION_NAME);
+    }
+
+    public enum UpdateMethod implements DescribedValue {
+        UPDATE_ONE("one", "Update One", "Updates only the first document that matches the query."),
+        UPDATE_MANY("many", "Update Many", "Updates every document that matches the query."),
+        UPDATE_FF_ATTRIBUTE("flowfile-attribute", "Use '" + ATTRIBUTE_MONGODB_UPDATE_MODE + "' flowfile attribute.",
+            "Use the value of the '" + ATTRIBUTE_MONGODB_UPDATE_MODE + "' attribute of the incoming flowfile. Acceptable values are 'one' and 'many'.");
+        private final String value;
+        private final String displayName;
+        private final String description;
+
+        UpdateMethod(final String value, final String displayName, final String description) {
+            this.value = value;
+            this.displayName = displayName;
+            this.description = description;
+        }
+
+        @Override
+        public String getValue() {
+            return value;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return displayName;
+        }
+
+        @Override
+        public String getDescription() {
+            return description;
+        }
+
+        public boolean matches(String value) {
+            return this.value.equals(value);
+        }
```

Comment:
This method should be removed, `enum` comparison can use standard `==` comparison.

Reply:
Thanks, I have updated it and also switched to using enum when reading processor property in Put processors